File: Data Structures & Algorithms/maximum-subarray/submission-2.java
class Solution {
    public int maxSubArray(int[] nums) {
        int total = 0;
        int cut = 0;
        int best = nums[0];
        for (int num : nums) {
            if (cut < 0) {
                total = total - cut;
                cut = 0;
            }
            total += num;
            cut += num;
            if (total > best) {
                best = total;
            }
        }

        return best;
    }
}

File: Data Structures & Algorithms/missing-number/submission-2.rs
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut xor = n as i32;
        for (i, num) in nums.iter().enumerate() {
            xor ^= (i as i32) ^ num;
        }
        xor
    }
}
